### PR TITLE
[#168169955] badge ServiceTabIcon

### PR DESCRIPTION
**Pull request by morrys**
The ServiceTabIcon badge shows the sum of the services displayed in the "Local" tab and those in the "National" tab:

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#168785770](https://www.pivotaltracker.com/story/show/168785770): Come CIT se sono nato all'estero, nel fac-simile del codice fiscale voglio trovare il mio luogo di nascita

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: ts/store/reducers/entities/services/index.ts [hunk 1]
```diff
@@ -225,4 +227,25 @@ export const notSelectedServicesSectionsSelector = createSelector(
...
+    isFirstVisibleServiceLoadCompletedSelector
+  ],
+  (
+    nationalService,
+    localService,
+    isFirstVisibleServicesLoadCompleted
+  ) => {
+    let result = 0;
+    if (isFirstVisibleServicesLoadCompleted) {
+      nationalService.length > 0 && nationalService.forEach(service => result += service.data.length)
```
Comment (francescopersico) [inline ts/store/reducers/entities/services/index.ts]: Is the length check needed? You are using forEach.

File: ts/store/reducers/entities/services/index.ts [hunk 2]
```diff
@@ -225,4 +227,25 @@ export const notSelectedServicesSectionsSelector = createSelector(
...
+    nationalServicesSectionsSelector,
+    selectedLocalServicesSectionsSelector,
+    isFirstVisibleServiceLoadCompletedSelector
+  ],
+  (
+    nationalService,
+    localService,
+    isFirstVisibleServicesLoadCompleted
+  ) => {
+    let result = 0;
```
Comment (francescopersico) [inline ts/store/reducers/entities/services/index.ts]: If possible try to avoid "let".

File: ts/store/reducers/entities/services/index.ts [hunk 3]
```diff
@@ -225,4 +227,25 @@ export const notSelectedServicesSectionsSelector = createSelector(
   }
 );
 
+// Get the number of visible services that are not yet marked as read
```
Comment (francescopersico) [inline ts/store/reducers/entities/services/index.ts]: Where are you checking the read state?

File: ts/store/reducers/entities/services/index.ts [hunk 4]
```diff
@@ -225,4 +227,25 @@ export const notSelectedServicesSectionsSelector = createSelector(
...
+export const servicesBadgeValueSelector = createSelector(
+  [
+    nationalServicesSectionsSelector,
+    selectedLocalServicesSectionsSelector,
```
Comment (francescopersico) [inline ts/store/reducers/entities/services/index.ts]: From the name i am quite sure this is not the selector you need.

File: ts/store/reducers/entities/services/index.ts (hunk 3 above)
Comment (morrys) [reply ts/store/reducers/entities/services/index.ts]: in this commit  https://github.com/teamdigitale/io-app/pull/1336/commits/2131cbba7eb43b67c2428d4d92304cb1564e4641 I added the state read check

File: ts/store/reducers/entities/services/index.ts (hunk 2 above)
Comment (morrys) [reply ts/store/reducers/entities/services/index.ts]: in this commit  https://github.com/teamdigitale/io-app/pull/1336/commits/2131cbba7eb43b67c2428d4d92304cb1564e4641 I removed the use of let

File: ts/store/reducers/entities/services/index.ts (hunk 1 above)
Comment (morrys) [reply ts/store/reducers/entities/services/index.ts]: in this commit https://github.com/teamdigitale/io-app/commit/2131cbba7eb43b67c2428d4d92304cb1564e4641 I no longer use the forEach function

File: ts/store/reducers/entities/services/index.ts [hunk 5]
```diff
@@ -225,4 +228,36 @@ export const notSelectedServicesSectionsSelector = createSelector(
   }
 );
 
+// Get the number of visible services that are not yet marked as read
```
Comment (Undermaken) [inline ts/store/reducers/entities/services/index.ts]:
```suggestion
// Get the sum of selected local services + national services that are not yet marked as read
```

**Pull request merged**